### PR TITLE
hevm: correctly handle c++ exceptions on darwin - fixes #248

### LINKED ISSUE #248: hevm: some MODEXP calls fail with libc++abi.dylib: terminating with uncaught exception of type int
Running any code that triggers the `modexp` precompile leads to:
```
hevm bc-test --file ~/dapphub/klab/evm-semantics/tests/ethereum-tests/BlockchainTests/GeneralStateTests/stCreate2/create2callPrecompiles_d5g0v0.json
create2callPrecompiles_d5g0v0_ConstantinopleFix libc++abi.dylib: terminating with uncaught exception of type int
[1]    83906 abort      hevm bc-test --file
```

### PR DESCRIPTION
**Pull request by desaperados**

### CODE REVIEW

File: src/hevm/hevm.cabal [hunk 1]
```diff
@@ -150,6 +150,8 @@ executable hevm
     -Wall -threaded -with-rtsopts=-N
   other-modules:
     Paths_hevm
+  if os(darwin)
+    ld-options: -Wl,-keep_dwarf_unwind
```
Comment (rainbreak) [inline src/hevm/hevm.cabal]: Can you explain a bit more?

Comment (desaperados) [reply src/hevm/hevm.cabal]: Yes, how to summarise - I spent a bit of time chasing this around :-) The issue turned out to be with some broken c++ exception handling on darwin, triggered by the try/catch behaviour in ethjet-ff.cc. These options are link time optimisation flags which allow fallback to dwarf exception tables, overriding the default GHC -no_compact_unwind which seems to cause the broken behaviour.

Comment (rainbreak) [reply src/hevm/hevm.cabal]: Wow ok.

**Comment by rainbreak** [comment]
Tests aren't being rerun, as hevm-compliance is being taken from the cache. Maybe the change to hevm.cabal isn't getting picked up by nix somehow. @desaperados did you rebuild with nix or just with cabal when you tested locally?

**Comment by rainbreak** [comment]
Oh I think it's fine because all the darwin tests are passing now.

**Comment by tabathap** [comment]
👌

**Pull request merged**